### ISSUE TITLE
Needs support for authentication

### ISSUE THREAD

**Issue by lexi-lambda** [jeapostrophe/mongodb#5]
It doesn't appear that this supports any form of authentication with Mongo? All drivers should implement an authentication mechanism using the standard [Connection String URI Format](http://docs.mongodb.org/manual/reference/connection-string/).

I've hand-rolled the authentication handshake myself using [`MONGODB-CR` authentication method](http://docs.mongodb.org/manual/core/authentication/#mongodb-cr-authentication), but that is no longer sufficient. It has been deprecated and replaced with the more complicated [`SCRAM-SHA-1` authentication method](http://docs.mongodb.org/manual/core/authentication/#scram-sha-1-authentication) with MongoDB 3.0.0, so it would be a good idea to implement that.

**Comment by jeapostrophe** [comment]
I'm not using mongodb anymore, so I have no intention of changing the code. I wrote this back around MongoDB 1.7 and haven't really touched it since. I'm happy to merge any pull requests or directly add you as a contributor.

**Comment by lexi-lambda** [comment]
Yep, that's fine—I think I talked to you about this package a long time ago, and you told me that then. I have a stub implementation of the new auth method, and if I finish it, I'll open a pull request. I'm mostly just leaving this issue here for documentation's sake.

**Comment by Clamwinds** [comment]
Any chance we could write a new mongodb package? For some reason im not able to get a list of collections of any of the databases I have.

**Comment by lexi-lambda** [comment]
I think this package could really use a maintainer, so if you’re interested, feel free. :wink: I tried to update it to work with the Mongo DB 3 SCRAM-SHA-1 auth scheme, but it’s really poorly documented, and I had trouble getting it to work. I haven’t completely given up on it, but it’s difficult enough that I’m unlikely to do it without a good chunk of free time and a good reason to do it.
